RHDEVDOCS-4893 - 4.13 CMO config map reference - autogenerated (#57142)
openshift/openshift-docs: monitoring/config-map-reference-for-the-cluster-monitoring-operator.adoc

|resources|*v1.ResourceRequirements|Defines resource requests and limits for the Alertmanager container.

> |secrets|[]string|Defines a list of secrets to be mounted into Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They are added as volumes named `secret-<secret-name>` and mounted at `/etc/alertmanager/secrets/<secret-name>` in the `alertmanager` container in the Alertmanager pods.

Reviewer: I think it should be the same as Line 118

```

> |secrets|[]string|Defines a list of secrets to be mounted into Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They are added as volumes named `secret-<secret-name>` and mounted at `/etc/alertmanager/secrets/<secret-name>` in the `alertmanager` container in the Alertmanager pods.

Reviewer: ```suggestion
|secrets|[]string|Defines a list of secrets that need to be mounted into the Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They will be added as volumes named secret-<secret-name> and mounted at /etc/alertmanager/secrets/<secret-name> within the 'alertmanager' container of the Alertmanager Pods.
```
I think it should be the same as Line 118

> |secrets|[]string|Defines a list of secrets to be mounted into Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They are added as volumes named `secret-<secret-name>` and mounted at `/etc/alertmanager/secrets/<secret-name>` in the `alertmanager` container in the Alertmanager pods.

Reviewer: ```suggestion
|secrets|[]string|Defines a list of secrets to be mounted into Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They are added as volumes named `secret-<secret-name>` and mounted at `/etc/alertmanager/secrets/<secret-name>` in the `alertmanager` container of the Alertmanager pods.
```

Author: fixed

|resources|*v1.ResourceRequirements|Defines resource requests and limits for the Alertmanager container.

> |secrets|[]string|Defines a list of secrets that need to be mounted into the Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They will be added as volumes named secret-<secret-name> and mounted at /etc/alertmanager/secrets/<secret-name> within the 'alertmanager' container of the Alertmanager Pods.

Reviewer: ```suggestion
|secrets|[]string|Defines a list of secrets to be mounted into Alertmanager. The secrets must reside within the same namespace as the Alertmanager object. They are added as volumes named `secret-<secret-name>` and mounted at `/etc/alertmanager/secrets/<secret-name>` in the `alertmanager` container of the Alertmanager pods.
```

Author: Changed per your suggestion

[options="header"]
|===
| Property | Type | Description
|nodeSelector|map[string]string|Defines the nodes on which the pods are scheduled.

|tolerations|[]v1.Toleration|Defines tolerations for the pods.

|===

== NodeExporterCollectorBuddyInfoConfig

=== Description

> The `NodeExporterCollectorBuddyInfoConfig` resource works as an on/off switch for the `buddyinfo` collector of the `node-exporter` agent. By default, the `buddyinfo` collector is disabled.

Author: Agree. I will add this to my list for changes to the source code.

|buddyinfo|link:#nodeexportercollectorbuddyinfoconfig[NodeExporterCollectorBuddyInfoConfig]|Defines the configuration of the `buddyinfo` collector, which collects statistics about memory fragmentation from the `node_buddyinfo_blocks` metric. This metric collects data from `/proc/buddyinfo`. Disabled by default.

|===

== NodeExporterCollectorCpufreqConfig

=== Description

The `NodeExporterCollectorCpufreqConfig` resource works as an on/off switch for the `cpufreq` collector of the `node-exporter` agent. By default, the `cpufreq` collector is disabled. Under certain circumstances, enabling the cpufreq collector increases CPU usage on machines with many cores. If you enable this collector and have machines with many cores, monitor your systems closely for excessive CPU usage.

Appears in: link:#nodeexportercollectorconfig[NodeExporterCollectorConfig]

[options="header"]
|===
| Property | Type | Description
|enabled|bool|A Boolean flag that enables or disables the `cpufreq` collector.

|===

== NodeExporterCollectorNetClassConfig

=== Description

> The `NodeExporterCollectorNetClassConfig` resource works as an on/off switch for the `netclass` collector of the `node-exporter` agent. By default, the `netclass` collector is enabled. If disabled, these metrics become unavailable: `node_network_info`, `node_network_address_assign_type`, `node_network_carrier`, `node_network_carrier_changes_total`, `node_network_carrier_up_changes_total`, `node_network_carrier_down_changes_total`, `node_network_device_id`, `node_network_dormant`, `node_network_flags`, `node_network_iface_id`, `node_network_iface_link`, `node_network_iface_link_mode`, `node_network_mtu_bytes`, `node_network_name_assign_type`, `node_network_net_dev_group`, `node_network_speed_bytes`, `node_network_transmit_queue_length`, `node_network_protocol_type`.

Author: Not sure if this can be achieved given that the source of the content is a comment in the source code. But I will add this to my list to investigate for the next release. Thanks!

[options="header"]
|===
| Property | Type | Description
|enabled|bool|A Boolean flag that enables or disables the `cpufreq` collector.

|===

== NodeExporterCollectorNetClassConfig

=== Description

The `NodeExporterCollectorNetClassConfig` resource works as an on/off switch for the `netclass` collector of the `node-exporter` agent. By default, the `netclass` collector is enabled. If disabled, these metrics become unavailable: `node_network_info`, `node_network_address_assign_type`, `node_network_carrier`, `node_network_carrier_changes_total`, `node_network_carrier_up_changes_total`, `node_network_carrier_down_changes_total`, `node_network_device_id`, `node_network_dormant`, `node_network_flags`, `node_network_iface_id`, `node_network_iface_link`, `node_network_iface_link_mode`, `node_network_mtu_bytes`, `node_network_name_assign_type`, `node_network_net_dev_group`, `node_network_speed_bytes`, `node_network_transmit_queue_length`, `node_network_protocol_type`.

Appears in: link:#nodeexportercollectorconfig[NodeExporterCollectorConfig]

[options="header"]
|===
| Property | Type | Description
|enabled|bool|A Boolean flag that enables or disables the `netclass` collector.

> |useNetlink|bool|A Boolean flag that activates the `netlink` implementation of the `netclass` collector. This implementation improves the performance of the `netclass` collector by omitting these metrics: `node_network_address_assign_type`, `node_network_name_assign_type`, `node_network_device_id`, `node_network_speed_bytes`. In addition, The `node_network_info` metric lacks the `duplex` label.

Reviewer 2: Do we need to hint that `netlink` is disabled by default? Although it's currently fixed in 4.14, it should be backported to 4.13 soon.
see from https://issues.redhat.com/browse/OCPBUGS-8282

Reviewer: @raptorsun ^

[options="header"]
|===
| Property | Type | Description
|enabled|bool|A Boolean flag that enables or disables the `tcpstat` collector.

|===

== NodeExporterConfig

=== Description

The `NodeExporterConfig` resource defines settings for the `node-exporter` agent.

Appears in: link:#clustermonitoringconfiguration[ClusterMonitoringConfiguration]

[options="header"]
|===
| Property | Type | Description
|collectors|link:#nodeexportercollectorconfig[NodeExporterCollectorConfig]|Defines which collectors are enabled and their additional configuration parameters.

|maxProcs|uint32|The target number of CPUs on which the Node Exporter's process will run. The default value here is `0`, which means that Node Exporter runs on all CPUs. If a kernel deadlock occurs or if performance degrades when reading from `sysfs` concurrently, you can change this value to `1`, which limits Node Exporter to running on one CPU. For nodes with a high CPU count, you can set the limit to a low number, which  saves resources by preventing Go routines from being scheduled to run on all CPUs. However, I/O performance degrades if the `maxProcs` value is set too low, and there are many metrics to collect.

> 

Reviewer 2: maxProcs is not on 4.13 currently.
FYI: https://issues.redhat.com//browse/MON-2894
